validphys2/src/validphys/results.py

```diff
@@ -1098,6 +1165,51 @@ def fits_chi2_table(
     return res
 
 
+@table
+def fits_nsigma_table(
+    fits_total_chi2_data,
+    fits_datasets_nsigma_table,
+    fits_groups_nsigma_table,
+    show_total: bool = False,
+):
+    """Show the nsigma of each and number of points of each dataset and experiment
+    of each fit, where experiment is a group of datasets according to the `experiment` key in
+    the PLOTTING info file, computed with the theory corresponding to the fit. Dataset that are not
+    included in some fit appear as `NaN`
+    """
+    lvs = fits_groups_nsigma_table.index
+    # The explicit call to list is because pandas gets confused otherwise
+    expanded_index = pd.MultiIndex.from_product((list(lvs), ["Total"]))
+    edf = fits_groups_nsigma_table.set_index(expanded_index)
+    ddf = fits_datasets_nsigma_table
+    dfs = []
+    # TODO: Better way to do the merge preserving the order?
+    for lv in lvs:
+        dfs.append(pd.concat((edf.loc[lv], ddf.loc[lv]), copy=False, axis=0))
+    if show_total:
+        total_points = np.array([total_chi2_data.ndata for total_chi2_data in fits_total_chi2_data])
+        total_chi = np.array(
+            [
+                (total_chi2_data.central_result - total_chi2_data.ndata)
+                / np.sqrt(2.0 * total_chi2_data.ndata)
+                for total_chi2_data in fits_total_chi2_data
+            ]
+        )
```

Comment:
If a list comprehesion needs more than a line, I think it is brtter to write the loop explictly.

In particular here you are doing twice the same loop (here and above)

Reply:
Ok, I added a single for loop, let me know what you think.